src/instructlab/model/serve.py

```diff
@@ -155,9 +164,29 @@ def serve(
             num_threads=num_threads,
         )
     elif backend == backends.VLLM:
+        # First Party
+        from instructlab.model.backends.vllm import contains_argument
+
         # Warn if unsupported backend parameters are passed
         warn_for_unsuported_backend_param(ctx)
 
+        if gpus:
```

Comment:
(Suggestion) This code is a bit convoluted. It would be helpful to see some test cases that demonstrate how this works. Ideally, something that would allow us to assert on `vllm_args` state past line 197. (May require some code extraction into a separate testable function.)

Reply:
I've added a bunch of tests.  Hopefully they are a good start.